Connection/Connection.php

```diff
@@ -337,14 +337,31 @@ public function disconnect()
 
     /**
      * Set socket.
+     * A wrapper can be registered to handle application layer transport
      *
-     * @param   string  $socket    Socket URI.
+     * @param   string  $socketUri    Socket URI.
      * @return  \Hoa\Socket
+     * @throws  \Hoa\Socket\Exception
      */
-    protected function setSocket($socket)
+    protected function setSocket($socketUri)
     {
+        $parsed = parse_url($socketUri);
+        if (null !== $wrapper =
+            Socket\Transport::getWrapper($parsed['scheme'])) {
+            $socket = call_user_func($wrapper, $socketUri);
```

Comment:
Because we are dealing with `callable` which can have different form :

* `[ $object, 'method ]`
* 'ClassName::staticMethod'
* function() {}

If we call directly $wrappers as a function, we are using `Closure` so we must update the type hinting in the registerWrapper.